src/Expr.cc

```diff
@@ -4294,13 +4294,15 @@ InExpr::InExpr(ExprPtr arg_op1, ExprPtr arg_op2)
 
 	if ( op1->GetType()->Tag() == TYPE_PATTERN )
 		{
-		if ( op2->GetType()->Tag() != TYPE_STRING )
+		if ( op2->GetType()->Tag() == TYPE_STRING || op2->GetType()->Tag() == TYPE_TABLE )
```

Comment:
Currently, this only checks if the second type is a table. This means that code like this:

```
local s1: set[count] = { 1, 5, 7 };
print ( /pat1/ in s1 ) ;
```

will happily run and print false.

I suspect we should catch this and do the same that happens when we, e.g., try to compare an integer with a table of patterns. In this case, the output of:

```
local s4: set[pattern] = { /pat1/ };
print (5 in s4);
```

is something along the lines of:

```
error in pattern and ./c.zeek, line 8: arithmetic mixed with non-arithmetic (pattern and 5)
error in ./c.zeek, line 8 and pattern: type mismatch (5 and pattern)
error in ./c.zeek, line 8: not an index type (5 in s4)
```

Reply:
Nice catch — I'll get on it.

Reply:
I've reworked the logic in the `InExpr` constructor a bit further so the pattern case falls through to the general type-checking at the end of the function, which catches the above. I've expanded the corresponding test case, too.